src/clarity-angular/datagrid/datagrid.spec.ts

```diff
@@ -157,13 +158,53 @@ export default function(): void {
             it("respects the trackBy option when using clrDgItems", function () {
                 this.context = this.create(Datagrid, TrackByTest);
                 let oldFirstRow = this.context.clarityElement.querySelector("clr-dg-row");
-                this.context.testComponent.items = [42];
+                this.context.testComponent.items = [ 42 ];
                 this.context.detectChanges();
                 let newFirstRow = this.context.clarityElement.querySelector("clr-dg-row");
                 expect(newFirstRow).toBe(oldFirstRow);
             });
         });
 
+        describe("Actionable rows", function () {
+            let context: TestContext<Datagrid, ActionableRowTest>;
+            let rowActionService: RowActionService;
+            let headActionOverflowCell: HTMLElement;
+            let actionOverflowCell: HTMLElement[];
+            let actionOverflow: HTMLElement[];
+
+            it("have cells for action overflows if there is at least one of them.", function () {
+                context = this.create(Datagrid, ActionableRowTest);
+                rowActionService = context.getClarityProvider(RowActionService);
+                context.detectChanges();
+                let datagridHead = context.clarityElement.querySelector(".datagrid-head");
+                headActionOverflowCell = datagridHead.querySelector(".datagrid-column.datagrid-row-actions");
+                actionOverflowCell = context.clarityElement.querySelectorAll("clr-dg-cell.datagrid-single-select");
+                actionOverflow = context.clarityElement.querySelectorAll("clr-dg-action-overflow");
+                expect(rowActionService.hasActionableRow).toBe(true);
+                expect(headActionOverflowCell).not.toBeNull();
+                expect(actionOverflowCell.length).toEqual(3);
+                expect(actionOverflow.length).toEqual(3);
+            });
+
+            it("have no cells for action overflows if there is none of them.", function () {
+                context = this.create(Datagrid, ActionableRowTest);
+                rowActionService = context.getClarityProvider(RowActionService);
+                context.testComponent.showIfGreaterThan = 10;
+                context.detectChanges();
+                actionOverflow = context.clarityElement.querySelectorAll("clr-dg-action-overflow");
+                expect(actionOverflow.length).toEqual(0);
+                context.detectChanges();
+                expect(rowActionService.hasActionableRow).toBe(false);
+                context.detectChanges();
```

Comment:
Why do you need these 2 extra change detections? That's surprising. You're not performing any new action, just doing`expect()`.